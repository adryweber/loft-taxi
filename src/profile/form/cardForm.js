import React from "react";
import Paper from "@material-ui/core/Paper";

import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Tooltip from "@material-ui/core/Tooltip";
import { MCIcon } from "loft-taxi-mui-theme";
import HelpOutlineIcon from "@material-ui/icons/HelpOutline";

import { makeStyles } from "@material-ui/core/styles";
import { useStyles as outStyle } from "../../shared/style";

const useStyles = makeStyles(theme => ({
    cardsWrapper: {
        width: "100%",
        display: "flex",
        justifyContent: "center",
        padding: `${theme.spacing(4)}px 0`
    },
    card: {
        position: "relative",
        width: 380,
        height: 200,
        padding: `${theme.spacing(2)}px ${theme.spacing(4)}px`,
        display: "flex",
        flexDirection: "column",
        borderRadius: 12,

        "& + &": {
            marginLeft: theme.spacing(6)
        }
    },
    shortField: {
        width: 130,
        position: "relative"
    },
    icoHelp: {
        position: "absolute",
        right: -40,
        bottom: 5
    },
    button: {
        width: 200
    }
}));

export const CardForm = () => {
    const classes = { ...useStyles(), ...outStyle() };

    const onSubmit = e => {
        // e.preventDefault();
        // setPage("profile");
    };

    return (
        <form onSubmit={onSubmit}>
            <div className={classes.cardsWrapper}>
                <Paper elevation={8} className={classes.card}>
                    <TextField
                        id="cardNumber"
                        label="Номер карты"
                        margin="normal"
                    />
                    <div className={classes.shortField}>
                        <TextField
                            id="expire"
                            label="Срок действия"
                            margin="normal"
                        />
                    </div>
                    <MCIcon />
                </Paper>
                <Paper elevation={8} className={classes.card}>
                    <TextField
                        id="holderName"
                        label="Имя владельца"
                        margin="normal"
                    />
                    <div className={classes.shortField}>
                        <TextField id="cvc" label="CVC" margin="normal" />
                        <Tooltip
                            title="3 последние цифры на оборотной стороне карты"
                            className={classes.icoHelp}
                        >
                            <HelpOutlineIcon color="primary" />
                        </Tooltip>
                    </div>
                </Paper>
            </div>

            <Button
                className={classes.button}
                variant="contained"
                color="primary"
                size="large"
            >
                Сохранить
            </Button>
        </form>
    );
};
