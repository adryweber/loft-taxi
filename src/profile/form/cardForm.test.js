import React from "react";
import { CardForm } from "./cardForm";
import renderer from "react-test-renderer";

it("Card Form renders correctly", () => {
    const tree = renderer.create(<CardForm />).toJSON();
    expect(tree).toMatchSnapshot();
});
