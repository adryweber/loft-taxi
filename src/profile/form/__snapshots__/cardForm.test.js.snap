// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Form renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="makeStyles-cardsWrapper-1"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation8 makeStyles-card-2 MuiPaper-rounded"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="cardNumber"
        >
          Номер карты
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="cardNumber"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
      <div
        className="makeStyles-shortField-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="expire"
          >
            Срок действия
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="expire"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root jss1"
        width="32px"
      >
        <img
          alt="mastercard"
          src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 131.39 86.9'%3E%3Cdefs%3E%3Cstyle%3E.a%7Bopacity:0;%7D.b%7Bfill:%23fff;%7D.c%7Bfill:%23ff5f00;%7D.d%7Bfill:%23eb001b;%7D.e%7Bfill:%23f79e1b;%7D%3C/style%3E%3C/defs%3E%3Ctitle%3EAsset 1%3C/title%3E%3Cg class='a'%3E%3Crect class='b' width='131.39' height='86.9'/%3E%3C/g%3E%3Crect class='c' x='48.37' y='15.14' width='34.66' height='56.61'/%3E%3Cpath class='d' d='M51.94,43.45a35.94,35.94,0,0,1,13.75-28.3,36,36,0,1,0,0,56.61A35.94,35.94,0,0,1,51.94,43.45Z'/%3E%3Cpath class='e' d='M120.5,65.76V64.6H121v-.24h-1.19v.24h.47v1.16Zm2.31,0v-1.4h-.36l-.42,1-.42-1h-.36v1.4h.26V64.7l.39.91h.27l.39-.91v1.06Z'/%3E%3Cpath class='e' d='M123.94,43.45a36,36,0,0,1-58.25,28.3,36,36,0,0,0,0-56.61,36,36,0,0,1,58.25,28.3Z'/%3E%3C/svg%3E"
          width="32px"
        />
      </span>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation8 makeStyles-card-2 MuiPaper-rounded"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="holderName"
        >
          Имя владельца
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="holderName"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
      <div
        className="makeStyles-shortField-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="cvc"
          >
            CVC
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="cvc"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
        <svg
          aria-describedby={null}
          aria-hidden="true"
          className="MuiSvgIcon-root makeStyles-icoHelp-4 MuiSvgIcon-colorPrimary"
          focusable="false"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="presentation"
          title="3 последние цифры на оборотной стороне карты"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
          />
        </svg>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Сохранить
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</form>
`;
